tests/prague/eip6206_jumpf/test_jumpf_execution.py

```diff
@@ -0,0 +1,176 @@
+                ),
+            ],
+        ),
+    )
+
+
+def test_jumpf_backward(
+    state_test: StateTestFiller,
+    eof_test: EOFTestFiller,
+):
+    """Tests JUMPF jumping backward"""
+    execute_tests(
+        state_test,
+        eof_test,
+        Container(
+            sections=[
+                Section.Code(
+                    code=Op.CALLF[2] + Op.SSTORE(0, 1) + Op.STOP,
+                    code_outputs=NON_RETURNING_SECTION,
+                    max_stack_height=2,
+                ),
+                Section.Code(
+                    code=Op.RETF,
+                ),
+                Section.Code(
+                    code=Op.JUMPF[1],
+                ),
+            ],
+        ),
+    )
+
+
+def test_jumpf_to_self(
+    state_test: StateTestFiller,
+    eof_test: EOFTestFiller,
+):
+    """Tests JUMPF jumping to self"""
+    execute_tests(
+        state_test,
+        eof_test,
+        Container(
+            sections=[
+                Section.Code(
+                    code=Op.SLOAD(0)
+                    + Op.ISZERO
+                    + Op.RJUMPI[1]
+                    + Op.STOP
+                    + Op.SSTORE(0, 1)
+                    + Op.JUMPF[0],
+                    code_outputs=NON_RETURNING_SECTION,
+                    max_stack_height=2,
+                )
+            ],
+        ),
+    )
+
+
+def test_jumpf_too_large(
+    state_test: StateTestFiller,
+    eof_test: EOFTestFiller,
+):
+    """Tests JUMPF jumping to a section outside the max section range"""
+    execute_tests(
+        state_test,
+        eof_test,
+        Container(
+            sections=[
+                Section.Code(
+                    code=Op.JUMPF[1025],
+                    code_outputs=NON_RETURNING_SECTION,
+                )
+            ],
+            validity_error=EOFException.UNDEFINED_EXCEPTION,
+        ),
+    )
+
+
+def test_jumpf_way_too_large(
+    state_test: StateTestFiller,
+    eof_test: EOFTestFiller,
+):
+    """Tests JUMPF jumping to uint64.MAX"""
+    execute_tests(
+        state_test,
+        eof_test,
+        Container(
+            sections=[
+                Section.Code(
+                    code=Op.JUMPF[0xFFFF],
+                    code_outputs=NON_RETURNING_SECTION,
+                )
+            ],
+            validity_error=EOFException.UNDEFINED_EXCEPTION,
+        ),
+    )
+
+
+def test_jumpf_to_nonexistent_section(
+    state_test: StateTestFiller,
+    eof_test: EOFTestFiller,
+):
+    """Tests JUMPF jumping to valid section number but where the section does not exist"""
+    execute_tests(
+        state_test,
+        eof_test,
+        Container(
+            sections=[
+                Section.Code(
+                    code=Op.JUMPF[5],
+                    code_outputs=NON_RETURNING_SECTION,
+                )
+            ],
+            validity_error=EOFException.UNDEFINED_EXCEPTION,
+        ),
+    )
+
+
+def test_callf_to_non_returning_section(
```

Comment:
this is very good. can clearly see the code and the error and nothing superfluous